src/negotiation/taker_negotiator.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { MockBitcoinWallet } from "../__mocks__/bitcoin_wallet";
+import { Cnd } from "../cnd";
+import { ComitClient } from "../comit_client";
+import { EthereumWallet } from "../ethereum_wallet";
+import {
+  // @ts-ignore: tslint does not know that this is actually ./__mocks__/maker_client
+  mockGetExecutionParams,
+  // @ts-ignore: tslint does not know that this is actually ./__mocks__/maker_client
+  mockGetOrderByTradingPair
+} from "./maker_client";
+import { TakerCriteria } from "./order";
+import { TakerNegotiator } from "./taker_negotiator";
+
+jest.mock("./maker_client");
+jest.mock("../ethereum_wallet");
+jest.mock("../cnd");
+
+const defaultCnd = new Cnd("");
+const defaultComitClient = new ComitClient(defaultCnd);
+defaultComitClient.bitcoinWallet = new MockBitcoinWallet();
+defaultComitClient.ethereumWallet = new EthereumWallet("");
+
+describe("Taker Negotiator", () => {
+  beforeEach(() => {
+    mockGetOrderByTradingPair.mockClear();
+    mockGetExecutionParams.mockClear();
+  });
+
+  it("Search an order and takes it", async () => {
+    const takerNegotiator = new TakerNegotiator(defaultComitClient, "");
+
+    // Define criteria for the order such as the assets and ledgers
+    // But also min/max amounts we are looking to trade
+    const takerCriteria: TakerCriteria = {
+      buy: {
+        ledger: "bitcoin",
+        asset: "bitcoin",
+        minNominalAmount: "2"
+      },
+      sell: {
+        ledger: "ethereum",
+        asset: "ether"
+      }
+    };
+
+    // Ask the maker if there are any orders that matches the criteria
+    const order = await takerNegotiator.getOrder(takerCriteria);
+
+    // We expect the returned order to be valid, if not
+    // This maker may be broken or malicious
```

Comment:
```suggestion
    // this maker may be broken or malicious.
```

🔧 Any particular reason why you start each line caps? Reads difficult if it is a sentence. (Note: same applies for comments below)

Reply:
I have no idea.